.design/requirements-575.md

```diff
@@ -0,0 +1,269 @@
+* *WebSocket Container* - The server-side, implementation-specific, container of server WebSocket Endpoints.
+* *WebSocket Session* - Description of a single client-server WebSocket connection.
+* *WebSocket Endpoint* - Handler of a WebSocket session lifecycle: open, close, error. The endpoint registers frame listeners when the session is opened. Can be used either on client or server side.
+* *WebSocket (Message) Handler* - Handler notified of WebSocket message reception. Can be used either on client or server side.
+
+## Problem Description
+
+Registering a WebSocket handler can currently be done by using a custom configuration of an HTTP servlet.
+The problem is that such configuration depends on the underlying HTTP server implementation.
+
+For example, the Eclipse Jetty server requires registering components before initializing an HTTP servlet to allow it to handle the WebSocket protocol.
+This requires to write code specific to Jetty in an OSGi HTTP whiteboard environment.
+There is also an issue between the Jetty WebSocket server components and the servlet context given to it by the Apache Felix HTTP Jetty implementation.
+
+As a result, the registration of WebSocket in the current state of the specification goes against the loose coupling concept usually provided by OSGi.
+
+What is needed is a WebSocket whiteboard specification moving specific configuration to the HTTP/WebSocket whiteboard implementation.
+
+Reusing parts, if not all, of the Jakarta WebSocket specification will ease the development of WebSocket handlers and avoid redefining existing APIs.
+It would also allow an easy reuse of existing Jakarta EE WebSocket handlers.
+
+## Note on *Partial* vs. *Whole* messages handlers
+
+The Jakarta WebSocket specification allows two kinds of message handlers which are described as:
+* *Whole* message handlers are notified by the container on arrival of a complete message (all frames recombined).
+* *Partial* message handlers are notified by the implementation when it is ready to deliver frames of a whole message.
+
+As it is explicitly stated in the JavaDoc part of the specification, the *partial* message handler is implementation-dependent.
+As a result, the OSGi WebSocket Whiteboard specification should reuse the `MessageHandler.Whole` type but ignore the `MessageHandler.Partial` one and the methods that accept it as argument.
+
+## Use Cases
+
+### UC1: Define and register a WebSocket server endpoint using annotations
+
+The code below is a WebSocket handler that we would want to register on the `/ws/work/{targetId}` endpoint.
+
+The OSGi WebSocket Whiteboard implementation is responsible to register an instance of this service as WebSocket endpoint and message handler, either directly or using a proxy endpoint class.
+
+```java
+@Component(service = WebSocketEndpointExample.class, scope = ServiceScope.PROTOTYPE)
+@WebSocketWhiteboardServletPattern("/ws/work/{targetId}") // See "WebSocket pattern description"
```

Comment:
This seems missing a `@ServerEndpoint` (what would carry the path already)

If an implementaion is interfacing with the WebSocket Spec one *might* want to resuse annotation and/or properties from the Servlet spec but having it explicit here seem not useful